v-next/hardhat/src/internal/builtin-plugins/network-manager/network-manager.ts

```diff
@@ -132,9 +136,14 @@ export class NetworkManagerImplementation {
     const createProvider = async (
       networkConnection: NetworkConnectionImplementation<ChainTypeT>,
     ): Promise<EthereumProvider> => {
-      if (resolvedNetworkConfig.type !== "http") {
-        /* eslint-disable-next-line no-restricted-syntax -- TODO implement EDR provider */
-        throw new Error("Only HTTP network is supported for now");
+      assertHardhatInvariant(
+        resolvedNetworkConfig.type === "edr" ||
+          resolvedNetworkConfig.type === "http",
+        `Invalid network type ${resolvedNetworkConfig.type}`,
+      );
+
+      if (resolvedNetworkConfig.type === "edr") {
+        return EdrProvider.create(resolvedNetworkConfig, { enabled: false });
```

Comment:
We need to expand the setup of tracing config and logging config. Or you know, have any.